File: src/main/java/net/pearadise/pearexpansion/block/enums/VerticalSlabType.java
package net.pearadise.pearexpansion.block.enums;

import net.minecraft.util.StringIdentifiable;

/**
 * Enum representing the different types of vertical slabs.
 * Each type corresponds to a specific orientation or a double slab.
 */
public enum VerticalSlabType implements StringIdentifiable {
    /**
     * Vertical slab facing north.
     */
    NORTH("north"),

    /**
     * Vertical slab facing south.
     */
    SOUTH("south"),

    /**
     * Vertical slab facing west.
     */
    WEST("west"),

    /**
     * Vertical slab facing east.
     */
    EAST("east"),

    /**
     * Double vertical slab.
     */
    DOUBLE("double");

    // The name of the vertical slab type.
    private final String name;

    /**
     * Constructor for the VerticalSlabType enum.
     *
     * @param name The name of the vertical slab type.
     */
    VerticalSlabType(final String name) {
        this.name = name;
    }

    /**
     * Returns the name of the vertical slab type as a string.
     *
     * @return The name of the vertical slab type.
     */
    public String toString() {
        return this.name;
    }

    /**
     * Returns the name of the vertical slab type as a string.
     * This method is required by the StringIdentifiable interface.
     *
     * @return The name of the vertical slab type.
     */
    @Override
    public String asString() {
        return this.name;
    }
}

File: src/main/kotlin/net/pearadise/pearexpansion/block/enums/VerticalSlabType.kt
package net.pearadise.pearexpansion.block.enums

import net.minecraft.util.StringIdentifiable

/**
 * Represents the possible types of vertical slabs.
 *
 * Each value defines the orientation of the slab or if it is a double slab.
 *
 * @property slabName The string name used for serialization and display.
 */
enum class VerticalSlabType(private val slabName: String) : StringIdentifiable {
    /** Vertical slab facing north. */
    NORTH("north"),

    /** Vertical slab facing south. */
    SOUTH("south"),

    /** Vertical slab facing west. */
    WEST("west"),

    /** Vertical slab facing east. */
    EAST("east"),

    /** Double vertical slab (fills the whole block). */
    DOUBLE("double");

    /**
     * Returns the string name of this slab type.
     *
     * @return The name of the slab type.
     */
    override fun toString(): String = slabName

    /**
     * Returns the string name of this slab type for registry and serialization.
     *
     * @return The name of the slab type.
     */
    override fun asString(): String = slabName
}